src/context/LoginContext.jsx

```diff
@@ -0,0 +1,42 @@
+import axios from "axios";
+
+import { createContext, useContext, useState } from "react";
+
+
+const LoginContext = createContext(null);
+
+const LoginProvider = ({ children }) => {
+ 
+ 
+  const [firstName, setFirstName] = useState("");
```

Comment:
Instead of half using three useState hooks. if possible combine into a single useState.
Example : - const [accountDetails,setaccountDetails] = useState({ firstName: '', lastName : '', encodedToken:null})

When updating firstName : - setaccountDetails({...accountDetails, firstName: userData.data.foundUser.firstName}.

Reply:
Yes It would be easier to maintain the states

Reply:
okay . I will do it for the time being